src/routes/[username]/[slug]/+page.server.ts

```diff
@@ -42,3 +48,14 @@ export const load = (async ({ params, locals, url }) => {
 		throw error(500, { message: e instanceof Error ? e.message : 'Something went wrong' });
 	}
 }) satisfies PageServerLoad;
+
+async function hash(ip: string) {
+	const data = new TextEncoder().encode(ip);
+	const buffer = await crypto.subtle.digest('SHA-256', data);
+
+	const hashHex = Array.from(new Uint8Array(buffer.slice(0, 8)))
+		.map((b) => b.toString(16).padStart(2, '0'))
+		.join('');
+
+	return parseInt(hashHex, 16);
+}
```

Comment:
```suggestion
async function hash(ip: string): Promise<bigint> {
  const data = new TextEncoder().encode(ip);
  const buffer = await crypto.subtle.digest('SHA-256', data);
  const view = new DataView(buffer);

  return view.getBigUint64(0);
}
```

This might be better